crates/spin-js-engine/src/lib.rs

```
@@ -696,6 +699,287 @@ fn open_kv(context: &Context, _this: &Value, args: &[Value]) -> Result<Value> {
+                }
+
+                let rows = context.array_value()?;
+
+                for row in result.rows.iter() {
+                    let temp = context.array_value()?;
+                    for val in row {
+                        let js_val = match val {
+                            mysql::DbValue::Boolean(v) => context.value_from_bool(*v)?,
+                            mysql::DbValue::Int8(v) => {
+                                context.value_from_i32(v.to_owned().into())?
+                            }
+                            mysql::DbValue::Int16(v) => {
+                                context.value_from_i32(v.to_owned().into())?
+                            }
+                            mysql::DbValue::Int32(v) => context.value_from_i32(*v)?,
+                            mysql::DbValue::Int64(v) => context.value_from_i64(*v)?,
+                            mysql::DbValue::Uint8(v) => {
+                                context.value_from_u32(v.to_owned().into())?
+                            }
+                            mysql::DbValue::Uint16(v) => {
+                                context.value_from_u32(v.to_owned().into())?
+                            }
+                            mysql::DbValue::Uint32(v) => context.value_from_u32(*v)?,
+                            mysql::DbValue::Uint64(v) => context.value_from_u64(*v)?,
+                            mysql::DbValue::Floating32(v) => {
+                                context.value_from_f64(v.to_owned().into())?
+                            }
+                            mysql::DbValue::Floating64(v) => context.value_from_f64(*v)?,
+                            mysql::DbValue::Str(v) => context.value_from_str(v)?,
+                            mysql::DbValue::Binary(v) => context.array_buffer_value(v)?,
+                            mysql::DbValue::DbNull => context.null_value()?,
+                            mysql::DbValue::Unsupported => context.undefined_value()?,
```

I'm thinking we should return an error here instead of using `undefined`.